contents/blog/temporal-exports.md

```diff
@@ -0,0 +1,131 @@
+
+2. **Postgres** - [health checks](https://github.com/PostHog/posthog/pull/16079) to ensure it is healthy before running workflows and [retry forever](https://github.com/PostHog/posthog/pull/16703) because we control everything related to those activities so failures are transient (other than bugs).
+
+3. **Destinations** - define which destination errors are [retriable and which aren’t](https://github.com/PostHog/posthog/pull/16814) (rule of thumb: user errors aren't, service errors are), [reset the iterator when a `JSONDecodeError` happens](https://github.com/PostHog/posthog/pull/16473) to keep the workflow going, and handle [query output](https://github.com/PostHog/posthog/pull/16247) errors from Snowflake.
+
+4. **Temporal** -  add [max retries and errors](https://github.com/PostHog/posthog/pull/15399) so it doesn’t retry forever, add [heartbeat API to track activity](https://github.com/PostHog/posthog/pull/16494) because we lose state on deployment causing exports to reset, and create an [endpoint](https://github.com/PostHog/posthog/pull/16181) for resetting a `BatchExportRun`.
+
+All of this and future improvements are possible because of the framework and services Temporal provides.
+
+### Making exports transparent (internally and externally)
+
+The second part of the move to Temporal was making processes transparent both internally and externally.
+
+As mentioned in the last section, Temporal creates transparency into the completions, errors, and latency of workflows. We also track bytes and records exported. These enable us to improve the export system further and faster than the old system.
+
+For users, the old system hides runs. Exports now [have their own page](app.posthog.com/batch_exports). On this page, users can:
+
+- Configure, start, [pause, unpause](https://github.com/PostHog/posthog/pull/16050), reset, [edit](https://github.com/PostHog/posthog/pull/16869), and [delete](https://github.com/PostHog/posthog/pull/16066) exports.
+
+- See the status, frequency, destination, and latest runs of each export.
+
+- Create historic exports (backfills) using [date range](https://github.com/PostHog/posthog/pull/16286).
+
+![Exports UI](../images/blog/temporal-exports/exports.png)
+
+The work to make exports transparent required us to improve encryption too. This included adding an [encryption codec](https://github.com/PostHog/posthog/pull/15566) because Temporal imports aren’t encrypted by default and our inputs can contain credentials. We also changed `BatchExportDestinations` to an `EncryptedJSONField` to ensure no encrypted credentials in the database and no credentials (neither encrypted nor unencrypted) in the logs.
```

Comment:
```suggestion
The work to make exports transparent required us to improve encryption, too. This included adding an [encryption codec](https://github.com/PostHog/posthog/pull/15566) because Temporal imports aren’t encrypted by default and our inputs can contain credentials. We also changed `BatchExportDestinations` to an `EncryptedJSONField` to ensure no encrypted credentials in the database and no credentials (neither encrypted nor unencrypted) in the logs.
```